catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/CategoryListFragment.java

```diff
@@ -238,47 +243,115 @@ public void onItemClick(CategoryListItem item) {
 				showSelectSpriteDialog();
 				break;
 			case CategoryListRVAdapter.DEFAULT:
-				((FormulaEditorFragment) getFragmentManager()
-						.findFragmentByTag(FormulaEditorFragment.FORMULA_EDITOR_FRAGMENT_TAG))
-						.addResourceToActiveFormula(item.nameResId);
-				getActivity().onBackPressed();
+				if (LIST_FUNCTIONS.contains(item.nameResId)) {
+					onUserListFunctionSelected(item);
+				} else {
+					addResourceToActiveFormulaInFormulaEditor(item);
+					getActivity().onBackPressed();
+				}
 				break;
 		}
 	}
 
+	private FormulaEditorFragment addResourceToActiveFormulaInFormulaEditor(CategoryListItem categoryListItem) {
+		FormulaEditorFragment formulaEditorFragment = ((FormulaEditorFragment) getFragmentManager()
+				.findFragmentByTag(FormulaEditorFragment.FORMULA_EDITOR_FRAGMENT_TAG));
+		formulaEditorFragment.addResourceToActiveFormula(categoryListItem.nameResId);
+		return formulaEditorFragment;
+	}
+
+	private void addResourceToActiveFormulaInFormulaEditor(CategoryListItem categoryListItem, UserList lastUserList) {
+		addResourceToActiveFormulaInFormulaEditor(categoryListItem).addUserListToActiveFormula(lastUserList.getName());
+		getActivity().onBackPressed();
+	}
+
+	private void onUserListFunctionSelected(CategoryListItem item) {
+		FragmentActivity activity = getActivity();
+		TextInputDialog.Builder builder = new TextInputDialog.Builder(activity);
+		final List<UserList> projectUserList =
+				ProjectManager.getInstance().getCurrentProject().getUserLists();
```

Comment:
According to @SimonStefan this is not yet possible for fragments. As it will still take time till the concerning PR is merged ProjectManager.getInstance() should be used until then.

Reply:
Ok, thanks!